#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Parquetry.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Roughcast.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/plants/protos/PottedTree.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/street_furniture/protos/Bench.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/BrushedAluminium.proto"

WorldInfo {
  info [
    "Elevator Safety Monitoring System"
    "EV Detection & Alarm Project"
  ]
  title "Elevator Lobby - EV Safety Monitor"
  basicTimeStep 16
  contactProperties [
    ContactProperties {
      material1 "puck_rim"
      material2 "floor"
      coulombFriction [
        0.5
      ]
    }
  ]
}
Viewpoint {
  orientation 0.04369251560913299 0.04539807133723934 -0.9980130155456916 1.6116269224335809
  position 0.39527870201981924 15.218788523426777 2.818642836299235
}
TexturedBackground {
  texture "empty_office"
}
TexturedBackgroundLight {
}
PointLight {
  attenuation 0 0 1
  color 1 0.98 0.9
  intensity 10
  location -1.8 -2 2.5
  castShadows TRUE
}
PointLight {
  attenuation 0 0 1
  color 1 0.98 0.9
  intensity 10
  location 1.8 -2 2.5
  castShadows TRUE
}
PointLight {
  attenuation 0 0 1
  color 0.9 0.95 1
  intensity 8
  location 0 2 2.8
  castShadows TRUE
}
Solid {
  translation 0 0 -0.01
  children [
    Shape {
      appearance Parquetry {
        type "chequered"
        textureTransform TextureTransform {
          scale 4 3
        }
      }
      geometry Box {
        size 10 8 0.02
      }
    }
  ]
  name "floor"
  boundingObject Box {
    size 10 8 0.02
  }
}
Solid {
  translation 0 -3.9 1.5
  children [
    Shape {
      appearance Roughcast {
        textureTransform TextureTransform {
          scale 4 1
        }
      }
      geometry Box {
        size 10 0.2 3
      }
    }
  ]
  name "back_wall"
  boundingObject Box {
    size 10 0.2 3
  }
}
Solid {
  translation -4.9 0 1.5
  children [
    Shape {
      appearance Roughcast {
        textureTransform TextureTransform {
          scale 3 2
        }
      }
      geometry Box {
        size 0.2 8 3
      }
    }
  ]
  name "left_wall"
  boundingObject Box {
    size 0.2 8 3
  }
}
Solid {
  translation 4.9 0 1.5
  children [
    Shape {
      appearance Roughcast {
        textureTransform TextureTransform {
          scale 3 2
        }
      }
      geometry Box {
        size 0.2 8 3
      }
    }
  ]
  name "right_wall"
  boundingObject Box {
    size 0.2 8 3
  }
}
Solid {
  translation 0 0 3
  children [
    Shape {
      appearance Roughcast {
        textureTransform TextureTransform {
          scale 5 5
        }
      }
      geometry Box {
        size 10 8 0.05
      }
    }
  ]
  name "ceiling"
}
DEF RestrictedZone Solid {
  translation 0 -2.5 0.001
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.1 0.1
        transparency 0.4
        roughness 0.5
        metalness 0
        emissiveColor 0.2 0 0
      }
      geometry Box {
        size 6 2 0.01
      }
    }
  ]
  name "restricted_zone"
}
Solid {
  translation 0 -1.5 0.002
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0.8 0
        roughness 0.6
        emissiveColor 0.2 0.15 0
      }
      geometry Box {
        size 6 0.1 0.01
      }
    }
  ]
  name "warning_stripe"
}
Solid {
  translation -1.8 -3.75 1.25
  children [
    Solid {
      translation 0 0 1.15
      children [
        Shape {
          appearance DEF BRUSHED_STEEL PBRAppearance {
            baseColor 0.7 0.72 0.75
            roughness 0.2
            metalness 0.8
          }
          geometry Box {
            size 1.6 0.1 0.1
          }
        }
      ]
      name "frame_top_1"
    }
    Solid {
      translation -0.75 0 0
      children [
        Shape {
          appearance USE BRUSHED_STEEL
          geometry Box {
            size 0.1 0.1 2.4
          }
        }
      ]
      name "frame_left_1"
    }
    Solid {
      translation 0.75 0 0
      children [
        Shape {
          appearance USE BRUSHED_STEEL
          geometry Box {
            size 0.1 0.1 2.4
          }
        }
      ]
      name "frame_right_1"
    }
  ]
  name "elevator_frame_1"
  boundingObject Group {
    children [
      Pose {
        translation -0.75 0 0
        children [
          Box {
            size 0.1 0.1 2.4
          }
        ]
      }
      Pose {
        translation 0.75 0 0
        children [
          Box {
            size 0.1 0.1 2.4
          }
        ]
      }
    ]
  }
}
Solid {
  translation 1.8 -3.75 1.25
  children [
    Solid {
      translation 0 0 1.15
      children [
        Shape {
          appearance USE BRUSHED_STEEL
          geometry Box {
            size 1.6 0.1 0.1
          }
        }
      ]
      name "frame_top_2"
    }
    Solid {
      translation -0.75 0 0
      children [
        Shape {
          appearance USE BRUSHED_STEEL
          geometry Box {
            size 0.1 0.1 2.4
          }
        }
      ]
      name "frame_left_2"
    }
    Solid {
      translation 0.75 0 0
      children [
        Shape {
          appearance USE BRUSHED_STEEL
          geometry Box {
            size 0.1 0.1 2.4
          }
        }
      ]
      name "frame_right_2"
    }
  ]
  name "elevator_frame_2"
  boundingObject Group {
    children [
      Pose {
        translation -0.75 0 0
        children [
          Box {
            size 0.1 0.1 2.4
          }
        ]
      }
      Pose {
        translation 0.75 0 0
        children [
          Box {
            size 0.1 0.1 2.4
          }
        ]
      }
    ]
  }
}
Robot {
  translation -2.1 -3.7 1.2
  children [
    SliderJoint {
      jointParameters JointParameters {
        axis -1 0 0
      }
      device [
        LinearMotor {
          name "door_motor_1L"
          maxVelocity 0.3
          minPosition -0.55
        }
        PositionSensor {
          name "door_sensor_1L"
        }
      ]
      endPoint DEF DOOR_1L Solid {
        children [
          Shape {
            appearance DEF DOOR_METAL BrushedAluminium {
              textureTransform TextureTransform {
                rotation 1.5708
              }
            }
            geometry Box {
              size 0.65 0.05 2.3
            }
          }
        ]
        name "door_1L_solid"
        boundingObject Box {
          size 0.65 0.05 2.3
        }
        physics Physics {
          density -1
          mass 50
        }
      }
    }
  ]
  name "elevator1_door_left"
  controller "<none>"
}
Robot {
  translation -1.5 -3.7 1.2
  children [
    SliderJoint {
      jointParameters JointParameters {
        axis 1 0 0
      }
      device [
        LinearMotor {
          name "door_motor_1R"
          maxVelocity 0.3
          maxPosition 0.55
        }
        PositionSensor {
          name "door_sensor_1R"
        }
      ]
      endPoint DEF DOOR_1R Solid {
        children [
          Shape {
            appearance USE DOOR_METAL
            geometry Box {
              size 0.65 0.05 2.3
            }
          }
        ]
        name "door_1R_solid"
        boundingObject Box {
          size 0.65 0.05 2.3
        }
        physics Physics {
          density -1
          mass 50
        }
      }
    }
  ]
  name "elevator1_door_right"
  controller "<none>"
}
Robot {
  translation 1.5 -3.7 1.2
  children [
    SliderJoint {
      jointParameters JointParameters {
        axis -1 0 0
      }
      device [
        LinearMotor {
          name "door_motor_2L"
          maxVelocity 0.3
          minPosition -0.55
        }
        PositionSensor {
          name "door_sensor_2L"
        }
      ]
      endPoint DEF DOOR_2L Solid {
        children [
          Shape {
            appearance USE DOOR_METAL
            geometry Box {
              size 0.65 0.05 2.3
            }
          }
        ]
        name "door_2L_solid"
        boundingObject Box {
          size 0.65 0.05 2.3
        }
        physics Physics {
          density -1
          mass 50
        }
      }
    }
  ]
  name "elevator2_door_left"
  controller "<none>"
}
Robot {
  translation 2.1 -3.7 1.2
  children [
    SliderJoint {
      jointParameters JointParameters {
        axis 1 0 0
      }
      device [
        LinearMotor {
          name "door_motor_2R"
          maxVelocity 0.3
          maxPosition 0.55
        }
        PositionSensor {
          name "door_sensor_2R"
        }
      ]
      endPoint DEF DOOR_2R Solid {
        children [
          Shape {
            appearance USE DOOR_METAL
            geometry Box {
              size 0.65 0.05 2.3
            }
          }
        ]
        name "door_2R_solid"
        boundingObject Box {
          size 0.65 0.05 2.3
        }
        physics Physics {
          density -1
          mass 50
        }
      }
    }
  ]
  name "elevator2_door_right"
  controller "<none>"
}
DEF Elevator1LEDSign Solid {
  translation -1.8 -3.62 2.65
  children [
    Shape {
      appearance DEF SIGN_FRAME PBRAppearance {
        baseColor 0.1 0.1 0.1
        roughness 0.4
        metalness 0.9
      }
      geometry Box {
        size 0.9 0.03 0.3
      }
    }
    Pose {
      translation 0 0.02 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 0.2
            metalness 0.8
          }
          geometry Box {
            size 0.82 0.01 0.22
          }
        }
      ]
    }
    Pose {
      translation -0.28 0.025 0
      children [
        Shape {
          appearance DEF LED_TEXT PBRAppearance {
            baseColor 0.2 0.8 1
            transparency 0.1
            emissiveColor 0.3 0.9 1
          }
        }
      ]
    }
    Pose {
      translation 0.35 0.025 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0.5
            emissiveColor 0 0.8 0.4
          }
          geometry Sphere {
            radius 0.02
          }
        }
      ]
    }
  ]
  name "elevator1_led_sign"
}
DEF Elevator2LEDSign Solid {
  translation 1.8 -3.62 2.65
  children [
    Shape {
      appearance USE SIGN_FRAME
      geometry Box {
        size 0.9 0.03 0.3
      }
    }
    Pose {
      translation 0 0.02 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 0.2
            metalness 0.8
          }
          geometry Box {
            size 0.82 0.01 0.22
          }
        }
      ]
    }
    Pose {
      translation -0.28 0.025 0
      children [
        Shape {
        }
      ]
    }
    Pose {
      translation 0.35 0.025 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0.5
            emissiveColor 0 0.8 0.4
          }
          geometry Sphere {
            radius 0.02
          }
        }
      ]
    }
  ]
  name "elevator2_led_sign"
}
Solid {
  translation -1.8 -4.3 1.2
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.3 0.3 0.35
        roughness 0.4
      }
      geometry Box {
        size 1.4 1 2.4
      }
    }
  ]
  name "elevator1_interior"
}
Solid {
  translation 1.8 -4.3 1.2
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.3 0.3 0.35
        roughness 0.4
      }
      geometry Box {
        size 1.4 1 2.4
      }
    }
  ]
  name "elevator2_interior"
}
Solid {
  translation -1.8 -2 2.95
  children [
    Shape {
      appearance PBRAppearance {
        emissiveColor 1 0.98 0.9
      }
      geometry Box {
        size 0.6 0.6 0.05
      }
    }
  ]
  name "light_fixture_1"
}
Solid {
  translation 1.8 -2 2.95
  children [
    Shape {
      appearance PBRAppearance {
        emissiveColor 1 0.98 0.9
      }
      geometry Box {
        size 0.6 0.6 0.05
      }
    }
  ]
  name "light_fixture_2"
}
Bench {
  translation 3.5 1 0
  rotation 0 0 1 3.14159
}
PottedTree {
  translation 0.89 2.66 0
}
Solid {
  translation -2 -1 0.15
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.6 0.6 0.65
        roughness 0.1
        metalness 0.8
      }
      geometry Cylinder {
        height 0.3
        radius 0.12
      }
    }
  ]
  name "obstacle_pillar_1"
  boundingObject Cylinder {
    height 0.3
    radius 0.12
  }
  physics Physics {
    density -1
    mass 10
  }
}
Solid {
  translation 2 -1 0.15
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.6 0.6 0.65
        roughness 0.1
        metalness 0.8
      }
      geometry Cylinder {
        height 0.3
        radius 0.12
      }
    }
  ]
  name "obstacle_pillar_2"
  boundingObject Cylinder {
    height 0.3
    radius 0.12
  }
  physics Physics {
    density -1
    mass 10
  }
}
Solid {
  translation 0 -0.5 0.1
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.5 0.4 0.3
        roughness 0.7
        name "cardboard"
      }
      geometry Box {
        size 0.2 0.2 0.2
      }
    }
  ]
  name "obstacle_box_1"
  boundingObject Box {
    size 0.2 0.2 0.2
  }
  physics Physics {
    density -1
    mass 2
  }
}
Solid {
  translation 1 -0.8 0.15
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.55 0.55 0.58
        roughness 0.2
        metalness 0.5
      }
      geometry Cylinder {
        height 0.3
        radius 0.1
      }
    }
  ]
  name "obstacle_pillar_3"
  boundingObject Cylinder {
    height 0.3
    radius 0.1
  }
  physics Physics {
    density -1
    mass 8
  }
}
DEF GreenPillar Solid {
  translation 0.35 2.66 0.5
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        roughness 0.3
        emissiveColor 0 0.5 0
      }
      geometry Cylinder {
        height 1
        radius 0.15
      }
    }
  ]
  name "green_pillar"
  boundingObject Cylinder {
    height 1
    radius 0.15
  }
}
DEF NonEVVehicle Robot {
  translation -1.42 0.9 0.9
  rotation 0 0 1 -1.5708
  children [
    Solid {
      translation 0 0 0.65
      children [
        Shape {
          appearance DEF ROBOT_METAL PBRAppearance {
            baseColor 0.7 0.75 0.8
            roughness 0.3
            metalness 0.8
          }
          geometry Box {
            size 0.2 0.22 0.25
          }
        }
        Solid {
          translation 0.11 0.05 0.02
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.1 0.1 0.1
                roughness 0.1
                metalness 0.9
              }
              geometry Box {
                size 0.01 0.05 0.02
              }
            }
          ]
          name "eye_left"
        }
        Solid {
          translation 0.11 -0.05 0.02
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.1 0.1 0.1
                roughness 0.1
                metalness 0.9
              }
              geometry Box {
                size 0.01 0.05 0.02
              }
            }
          ]
          name "eye_right"
        }
      ]
      name "head"
    }
    Solid {
      translation 0 0 0.2
      children [
        Shape {
          appearance USE ROBOT_METAL
          geometry Box {
            size 0.25 0.45 0.6
          }
        }
        Solid {
          translation -0.13 0 0.1
          children [
            Shape {
              appearance DEF GLOW_ORANGE PBRAppearance {
                baseColor 1 0.5 0
                transparency 0.1
                emissiveColor 1 0.3 0
              }
              geometry Box {
                size 0.05 0.25 0.35
              }
            }
            PointLight {
              attenuation 0 0 1
              color 1 0.4 0
              intensity 4
              location -0.2 0 0
              radius 2
            }
          ]
          name "back_power_pack"
        }
      ]
      name "torso"
    }
    Solid {
      translation 0 -0.28 0.2
      children [
        Shape {
          appearance USE ROBOT_METAL
          geometry Cylinder {
            height 0.6
            radius 0.06
          }
        }
        Solid {
          translation 0 0 -0.35
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.2 0.2 0.2
                roughness 0.5
              }
              geometry Sphere {
                radius 0.07
              }
            }
          ]
          name "hand_right"
        }
      ]
      name "arm_right"
    }
    Solid {
      translation 0 0.28 0.2
      children [
        Shape {
          appearance USE ROBOT_METAL
          geometry Cylinder {
            height 0.6
            radius 0.06
          }
        }
        Solid {
          translation 0 0 -0.35
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.2 0.2 0.2
                roughness 0.5
              }
              geometry Sphere {
                radius 0.07
              }
            }
          ]
          name "hand_left"
        }
      ]
      name "arm_left"
    }
    Solid {
      translation 0 -0.12 -0.45
      children [
        Shape {
          appearance USE ROBOT_METAL
          geometry Box {
            size 0.15 0.15 0.8
          }
        }
        Solid {
          translation 0.05 0 -0.4
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.1 0.1 0.1
              }
              geometry Box {
                size 0.25 0.16 0.1
              }
            }
          ]
          name "foot_right"
        }
      ]
      name "leg_right"
    }
    Solid {
      translation 0 0.12 -0.45
      children [
        Shape {
          appearance USE ROBOT_METAL
          geometry Box {
            size 0.15 0.15 0.8
          }
        }
        Solid {
          translation 0.05 0 -0.4
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.1 0.1 0.1
              }
              geometry Box {
                size 0.25 0.16 0.1
              }
            }
          ]
          name "foot_left"
        }
      ]
      name "leg_left"
    }
  ]
  name "NonEVVehicle"
  boundingObject Box {
    size 0.4 0.6 1.8
  }
  physics Physics {
    density -1
    mass 80
    centerOfMass [
      0 0 0
    ]
  }
  controller "<none>"
}
DEF ElectricVehicle Robot {
  translation -0.329996 2.36 0.13
  rotation 0 0 1 -1.5708
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.05 0.4 0.05
        roughness 0.2
        metalness 0.7
        name "evGreenPaint"
      }
      geometry Box {
        size 0.6 0.4 0.25
      }
    }
    Solid {
      translation 0 0 0.2
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            transparency 0.1
            emissiveColor 0 1 0
          }
          geometry Box {
            size 0.5 0.35 0.1
          }
        }
      ]
      name "ev_marker_top"
    }
    Solid {
      translation 0.35 0 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            emissiveColor 0 1 0
          }
          geometry Box {
            size 0.05 0.35 0.2
          }
        }
      ]
      name "ev_marker_front"
    }
    Solid {
      translation -0.35 0 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            emissiveColor 0 1 0
          }
          geometry Box {
            size 0.05 0.35 0.2
          }
        }
      ]
      name "ev_marker_back"
    }
    Solid {
      translation 0 0.22 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            emissiveColor 0 1 0
          }
          geometry Box {
            size 0.5 0.05 0.2
          }
        }
      ]
      name "ev_marker_left"
    }
    Solid {
      translation 0 -0.22 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            emissiveColor 0 1 0
          }
          geometry Box {
            size 0.5 0.05 0.2
          }
        }
      ]
      name "ev_marker_right"
    }
    Solid {
      translation 0.2 0.22 -0.1
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.1 0.1 0.1
            roughness 0.7
          }
          geometry Cylinder {
            height 0.1
            radius 0.16
          }
        }
      ]
      name "wheel_fr"
    }
    Solid {
      translation 0.2 -0.22 -0.1
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.1 0.1 0.1
            roughness 0.7
          }
          geometry Cylinder {
            height 0.1
            radius 0.16
          }
        }
      ]
      name "wheel_fl"
    }
    Solid {
      translation -0.2 0.22 -0.1
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.1 0.1 0.1
            roughness 0.7
          }
          geometry Cylinder {
            height 0.1
            radius 0.16
          }
        }
      ]
      name "wheel_rr"
    }
    Solid {
      translation -0.2 -0.22 -0.1
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.1 0.1 0.1
            roughness 0.7
          }
          geometry Cylinder {
            height 0.1
            radius 0.16
          }
        }
      ]
      name "wheel_rl"
    }
    Solid {
      translation 0 0 0.15
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.2 0.2 0.8
            emissiveColor 0.1 0.1 0.4
          }
          geometry Box {
            size 0.3 0.2 0.05
          }
        }
      ]
      name "battery"
    }
  ]
  name "ElectricVehicle"
  boundingObject Box {
    size 0.6 0.4 0.25
  }
  physics Physics {
    density -1
    mass 5
  }
  controller "<none>"
}
DEF WallAlarmLight Robot {
  translation -3.5 -3.5 2.2
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.1 0.1 0.1
        roughness 0.3
        metalness 0.8
      }
      geometry Box {
        size 0.3 0.15 0.3
      }
    }
    DEF ALARM_LIGHT_VISUAL Pose {
      translation 0 0.08 0
      children [
        Shape {
          appearance DEF ALARM_LIGHT_APPEARANCE PBRAppearance {
            baseColor 0.3 0 0
            roughness 0.3
            metalness 0
          }
          geometry Sphere {
            radius 0.1
          }
        }
      ]
    }
    PointLight {
      attenuation 0 0 1
      color 1 0 0
      intensity 0
      location 0 0.15 0
      radius 5
    }
  ]
  name "wall_alarm_light"
  controller "<none>"
}
DEF CameraAlarmLight Robot {
  translation 3.5 -3.5 2.2
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.1 0.1 0.1
        roughness 0.3
        metalness 0.8
      }
      geometry Box {
        size 0.3 0.15 0.3
      }
    }
    DEF CAMERA_LIGHT_VISUAL Pose {
      translation 0 0.08 0
      children [
        Shape {
          appearance DEF CAMERA_LIGHT_APPEARANCE PBRAppearance {
            baseColor 0 0 0.3
            roughness 0.3
            metalness 0
          }
          geometry Sphere {
            radius 0.1
          }
        }
      ]
    }
    PointLight {
      attenuation 0 0 1
      color 0 0.5 1
      intensity 0
      location 0 0.15 0
      radius 5
    }
  ]
  name "camera_alarm_light"
  controller "<none>"
}
DEF PatrolRobot E-puck {
  translation 3.13 -1.72 0
  rotation 0 0 1 3.14159
  name "camera_monitor"
  controller "epuck_fsm_controller"
  version "2"
  camera_width 128
  camera_height 128
  turretSlot [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.1 0.1
        emissiveColor 0.3 0 0
      }
      geometry Cylinder {
        height 0.01
        radius 0.025
      }
    }
  ]
}
DEF PatrolRobot2 E-puck {
  translation -3.13 -1.72 0
  rotation 0 0 1 -0.13
  name "camera_monitor_2"
  controller "epuck_camera_stationary"
  version "2"
  camera_width 128
  camera_height 128
  emitter_channel 2
  turretSlot [
    Emitter {
      name "cam2_emitter"
      channel 2
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0.1 0.1 0.8
        emissiveColor 0 0 0.3
      }
      geometry Cylinder {
        height 0.01
        radius 0.025
      }
    }
  ]
}
DEF VPE_Supervisor Robot {
  translation 0 0 0.01
  children [
    Emitter {
      channel 1
    }
    Receiver {
      channel 2
    }
  ]
  name "vpe_supervisor"
  controller "vpe_supervisor"
  supervisor TRUE
}
Solid {
  translation 0 -3.76 0.8
  children [
    Shape {
      appearance PBRAppearance {
        roughness 0.4
      }
      geometry Box {
        size 1 0.02 0.4
      }
    }
    Solid {
      translation 0 -0.02 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.9 0.1 0.1
            emissiveColor 0.3 0 0
          }
          geometry Box {
            size 0.95 0.01 0.35
          }
        }
      ]
      name "sign_background"
    }
  ]
  name "warning_sign"
}
Solid {
  translation -1.8 -3.65 2.55
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.1 0.1 0.1
        emissiveColor 0.8 0.4 0
      }
      geometry Box {
        size 0.3 0.02 0.15
      }
    }
  ]
  name "elevator1_indicator"
}
Solid {
  translation 1.8 -3.65 2.55
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.1 0.1 0.1
        emissiveColor 0.8 0.4 0
      }
      geometry Box {
        size 0.3 0.02 0.15
      }
    }
  ]
  name "elevator2_indicator"
}
Solid {
  translation -2 1.5 0.15
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.5 0.5 0.55
        roughness 0.6
      }
      geometry Cylinder {
        height 0.3
        radius 0.15
      }
    }
  ]
  name "pillar1"
  boundingObject Cylinder {
    height 0.3
    radius 0.15
  }
}
Solid {
  translation 2 -0.5 0.15
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.5 0.5 0.55
        roughness 0.6
      }
      geometry Cylinder {
        height 0.3
        radius 0.15
      }
    }
  ]
  name "pillar2"
  boundingObject Cylinder {
    height 0.3
    radius 0.15
  }
}
